content/kubernetes/deployment/deployment-options.md

```diff
@@ -0,0 +1,53 @@
+---
+Title: Flexible deployment options
+linktitle: Deployment options
+description: Redis Enterprise for Kubernetes allows you to deploy to multiple namespaces. This article describes flexible deployment options you can use to meet your specific needs. 
+weight: 12
+alwaysopen: false
+categories: ["Platforms"]
+aliases: [
+    /kubernetes/deployment/deployment-options.md,
+    /kubernetes/deployment/deployment-options/,
+]
+
+---
+Redis Enterprise for Kubernetes can be deployed in several different ways depending on your database needs.
+
+A single Redis Enterprise cluster (REC) can manage multiple Redis Enterprise database objects (REDB) in multiple namespaces within the same Kubernetes cluster.
+
+## Single REC and single namespace (one-to-one)
+
+The standard and simplest way is with your Redis Enterprise cluster (REC) and Redis Enterprise databases (REDB) residing within the same namespace. No additional configuration is required for this, since there is no communication required to cross namespaces. See [Deploy Redis Enterprise Software on Kubernetes]{{<relref "/kubernetes/deployment/quick-start.md">}}.
+
+![One-to-one deployment option](/images/platforms/k8s-deploy-one-to-one.png)
+
+## Single REC and multiple namespaces (one-to-many)
+
+A single Redis Enterprise cluster (REC) can manage multiple Redis Enterprise database objects (REDB) in multiple namespaces within the same Kubernetes cluster. See [Manage databases in multiple namespaces]({{<relref "/kubernetes/re-clusters/multi-namespace.md">}}) for more information.
+
+![one-to-many deployment option](/images/platforms/k8s-deploy-one-to-many.png)
+
+## Multiple RECs and multiple namespaces (many-to-many)
+
+A single Kubernetes cluster can contain multiple Redis Enterprise clusters (REC), as long as they reside in different namespaces. Each namespace can host only one REC.
+
+You have the flexibility to create databases in separate namespaces, or in the same namespace as the REC, or combine any of the supported deployment options above. This configuration is ideal for running a variety of applications in the same Kubernetes cluster.
+
+See [Manage databases in multiple namespaces]({{<relref "/kubernetes/re-clusters/multi-namespace.md">}}) for more information.
+
+
+![many-to-many deployment option](/images/platforms/k8s-deploy-many-to-many.png)
+
+## Unsupported deployment patterns
+
+### Cross-cluster operations
+
+Redis Enterprise for Kubernetes does not support operations that cross Kubernetes clusters. Redis Enterprise clusters (REC) work inside a single K8s cluster. Crossing clusters could result in functional and security issues.
+
+![cross cluster deployment antipattern](/images/platforms/k8s-deploy-cross-namespaces.png)
```

Comment:
TBH, I'm not sure what "cross-cluster operations" and "crossing clusters" mean here. Do you refer to some specific unsupported deployment pattern we've seen customers attempt?

Reply:
This is referring to the limitation that all these RECs and REDBs have to reside in the same K8s cluster. You can't have an REC in one k8s cluster monitor a REDB in a different K8s. You can cross namespaces, but not K8s clusters.

Reply:
No argument about this limitation. The point I was trying to make, is that the REC/REDB APIs don't expose anything that can look like a cross-cluster operation, so I'm not sure it's correct to say that these operations are "not supported" or "could result in..." since such operations merely don't exist within our APIs (unless I'm unaware of some bizzare configurations that customers have attempted and expected to link REC/REDBs across clusters?)

Reply:
Oh so you are saying it's not even possible to do this?

Reply:
Right. All references between objects are by name thus local to the cluster.
E.g., an REDB definition references the REC in which it's created like this:
```yaml
spec:
  redisEnterpriseCluster:
    name: re-cluster
```
And there's no way to reference a REC from another cluster.

Reply:
One notable exception to this is the upcoming AA controller, which does have some aspects of managing a multi-cluster entity. But that still shouldn't appear in the documentation.

Reply:
ok. I think I'll leave it as is for now and update it when we release the AA controller for better clarity. In general it's to warn customers when they are designing their deployment not to plan to cross clusters.